pkg/migrations/backfill.go

```diff
@@ -51,11 +50,38 @@ func backfill(ctx context.Context, conn *sql.DB, table *schema.Table, cbs ...Cal
 	return nil
 }
 
+// checkBackfill will return an error if the backfill operation is not supported.
+func checkBackfill(table *schema.Table) error {
+	col := getBackfillColumn(table)
+	if col == nil {
+		return BackfillNotPossible{Table: table.Name}
+	}
+
+	return nil
+}
+
+func getBackfillColumn(table *schema.Table) *schema.Column {
+	pks := table.GetPrimaryKey()
+	if len(pks) == 1 {
+		return pks[0]
+	}
+
+	// If there is no primary key, look for a unique not null column
+	for _, col := range table.Columns {
+		if col.Unique && !col.Nullable {
+			return &col
+		}
+	}
+
+	// no suitable column found
+	return nil
+}
```

Comment:
These two could be methods on `table` and moved alongside the other schema methods like `GetPrimaryKey`.

Reply:
I thought about it, but the domain knowledge of backfill is in this package and not that one, so it felt a bit weird